### DISCUSSION TITLE
How can I take a screen shot with ansi colors and pipe the result into `ansi2html`?

### DISCUSSION THREAD

**Discussion by David-Else** [kovidgoyal/kitty#8084] [Q&A] [answered]
Hi, I want to take screen shots of the entire kitty screen rather than just the current selection and pipe the results through `ansi2html` to use it as a screenshot on a website.

```sh
map alt+s               copy_ansi_to_clipboard
```
This works for the selection, but how do I pipe the result into `ansi2html`, and then maybe into `wl-copy`? I would like to end up with the output from `ansi2html` in the clipboard. I need to do `wl-paste | ansi2html` afterwards at the moment.
```sh
map alt+s               launch --type background --stdin-source=@screen copy_ansi_to_clipboard
```
This takes the entire screen, but there are no colors.

**Comment by kovidgoyal** [comment]
Write a script that does whatever you want and then

```
map alt+s launch --type background --stdin-source=@screen sh /path/to/your/script
```

**Reply by David-Else** [reply]
If I do as you say and make my script:
```sh
#!/bin/bash
cat - >~/Downloads/texttest
```
Then I do NOT have any ANSI escape codes in the resulting file. I need them to take a color screen shot. Thanks.

**Reply by kovidgoyal** [reply, marked as answer]
https://sw.kovidgoyal.net/kitty/launch/#cmdoption-launch-stdin-add-formatting

**Reply by David-Else** [reply]
Thanks!
